### PR TITLE
orders controller,order cum user associations modified and reset appl…

### PR DESCRIPTION
**Pull request by aleemchishti**
…ication layout file

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@ class OrdersController < ApplicationController
 	before_action :redirect_if_traveler, only: %i[new]
 
 	def index
-		@orders = current_user.orders 
+		if current_user.sender?
+		@orders = current_user.s_orders 
+		else
+		@orders = current_user.t_orders
+		end
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb]: use one line if condition in ruby something like below
`true? ? 'yes' : 'no'`

File: app/controllers/orders_controller.rb [hunk 2]
```diff
@@ -36,7 +35,7 @@ def edit
 	end
 
 	def update 
-	 	if current_user.sender
+	 	if current_user.sender?
 			@order = Order.find_by(id: params[:id])
```
Comment (momnanaseer) [inline app/controllers/orders_controller.rb]: move this line to before_action

File: app/models/user.rb [hunk 3]
```diff
@@ -5,6 +5,9 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
   
   enum role: [:sender, :traveler]
-  has_many :journey
-  has_many :orders
+  
+  has_many :journeys
+
+  has_many :s_orders, foreign_key: :sender_id, class_name: 'Order'
+  has_many :t_orders, foreign_key: :traveler_id, class_name: 'Order'
```
Comment (momnanaseer) [inline app/models/user.rb:12]: can you please explain this logic to me?

File: app/views/orders/_form.html.erb [hunk 4]
```diff
@@ -1,10 +1,9 @@
-<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">  
-<%= form_with(model: order) do |form| %>
+ <%= form_with(model: order) do |form| %>
 
   <br>
   <div class="form-group">
       <%= form.label :travelers, class:'form-label mb-2 mt-3' %><br />
-      <%= form.select :travelers,User.traveler.pluck(:first_name),{prompt:'Select'},{class: "form-control"}%>
+      <%= form.select :traveler_id,User.traveler.map{|u| [u.first_name, u.id]},{prompt:'Select'},{class: "form-control"}%>
```
Comment (momnanaseer) [inline app/views/orders/_form.html.erb]: use instance variable here, and query in controller

File: app/controllers/orders_controller.rb (hunk 1 above)
Comment (aleemchishti) [reply app/controllers/orders_controller.rb]: done

File: app/controllers/orders_controller.rb (hunk 2 above)
Comment (aleemchishti) [reply app/controllers/orders_controller.rb]: done

File: app/models/user.rb (hunk 3 above)
Comment (aleemchishti) [reply app/models/user.rb:12]: Using multiple foreign keys in order table. traveler_id and sender_id and both are coming from user table. There was no space for self joins so i had to apply this technique. And it is working now. Basically its representing has_many orders

**Pull request merged**